dataprofiler/data_readers/csv_data.py

```diff
@@ -96,29 +96,150 @@ def _determine_delimiter_of_str(data_as_str):
         :rtype: str or None
         """
         sniffer = csv.Sniffer()
-        sniffer.preferred = [',', '\t', ';']  # removes ' ', ';' from preferred
+        sniffer.preferred = [',', '\t', ';']  # removes ' ', ':' from preferred
         try:
             dialect = sniffer.sniff(data_as_str)
         except csv.Error as exc:
             return None
         return dialect.delimiter
 
     @staticmethod
-    def _determine_has_header(data_as_str):
-        """Automatically checks if the string has a header."""
-        sniffer = csv.Sniffer()
-        try:
-            has_header = sniffer.has_header(data_as_str)
-        except csv.Error:
-            has_header = 0
-        return 0 if has_header else None
+    def _guess_header_row(data_as_str, suggested_delimiter=None,
+                          diff_thresh=0.1, none_thresh=0.5, str_thresh=0.9):
+        """
+        This function attempts to select the best row for which a header would be valid.
+        
+        :param data_as_str: Single string containing rows (lines seperated by "\n")
+        :type data_as_str: str
+        :param suggested_delimiter: Delimiter suggested to use when trying to find the header
+        :type suggested_delimiter: str
+        :param diff_threshold: Max percent difference in cell types between rows allowed 
+        :type diff_threshold: float    
+        :param none_thresh: Max percent difference number of none values allowed
+        :type none_thresh: float    
+        :param str_thresh: Min percent of strings (omitting none) in row to be a header
+        :type str_thresh: float
+        
+        :return: index for row estimated to be the last valid header
+        :type: int
+        """
+        
+        # Catch base cases
+        if not data_as_str or len(data_as_str) == 0:
+            return None
+    
+        # Ensure no "None" delimiter
+        delimiter = suggested_delimiter
+        if not delimiter:
+            delimiter = ','
+    
+        # Ensure no empty last col and only 2 cols
+        empty_last_col_flag = True
+        for row in data_as_str.split('\n'):
+            last_cell = row.split(delimiter)[-1]
+            if len(row) == 2 and last_cell != None \
+            and len(last_cell) > 0:
+                empty_last_col_flag = False
+        if empty_last_col_flag:
```

Comment:
I'm concerned that this just defaults to `,` bc the last col. This seems. like a delimiter check and to me doesn't make sense here.

I can see this being temp as delimiter check gets updated. However, defaulting to `,` seems dangerous as that could just be completely wrong.

It shouldn't even hit this function if a delimiter can't be detected or is incorrectly detected.